test/test_factory.py

```diff
@@ -172,6 +173,27 @@ def _(loc, scale, value):
     check_funsor(actual, {"i": Bint[3]}, Real)
 
 
+@pytest.mark.skipif(get_backend() != "torch", reason="requires nn.Module")
+def test_nn_module():
+    import torch
+
+    class Matmul(torch.nn.Module):
+        def forward(
+            self,
+            x: Funsor,
+            y: Funsor,
+            i: Bound,
+        ) -> Fresh[lambda x: x]:
```

Comment:
Now that we have a `Dependent` I guess in a separate PR we should update the syntax of `make_funsor` to use `Dependent` rather than `Fresh` for return types in `make_funsor`.

Reply:
Sounds reasonable. I guess we can make `Fresh` subclass `Dependent` when introducing fresh variables.